tests/backends/postgresql/tests.py

```diff
@@ -407,6 +437,27 @@ def test_connect_role(self):
         finally:
             new_connection.close()
 
+    def test_custom_init_connection_role(self):
+        from django.db.backends.postgresql.base import DatabaseWrapper
+
+        class CustomDatabaseWrapper(DatabaseWrapper):
+            @classmethod
+            def init_connection_role(cls, connection, ops, role_name):
+                return False
+
+        custom_role = "django_nonexistent_role"
+        new_connection = no_pool_connection()
+        try:
+            new_connection.connect()
```

Comment:
Similarly to above, could we use `self.addCleanup(connection.close)` here instead of the try-finally block?